crates/sc-consensus-subspace-rpc/src/lib.rs

```diff
@@ -527,6 +528,18 @@ where
     }
 
     async fn records_roots(&self, segment_indexes: Vec<u64>) -> RpcResult<Vec<Option<Sha256Hash>>> {
+        if segment_indexes.len() > MAX_SEGMENT_INDEXES_PER_REQUEST {
+            error!(
+                "segment_indexes length exceed the limit: {} ",
+                segment_indexes.len()
+            );
+
+            return Err(JsonRpseeError::Custom(
+                "Internal error during `records_root` call: segment_indexes length exceed the limit"
```

Comment:
It must be clear what the request it was on the other side anyway and knowing the limit would help too:
```suggestion
                format!("segment_indexes length exceed the limit {MAX_SEGMENT_INDEXES_PER_REQUEST}")
```